### PR TITLE
agent: read full data from ctl and tty channels

### LINKED ISSUE #201: failed to read more than 4096 bytes from ctl
a new method to read all bytes from ctl and tty should be implemented to avoid errors reading more than 4096 bytes

### PR DESCRIPTION
**Pull request by devimc**
this patch is to read all the data sent by the proxy and to avoid
error reading more than 4096 bytes

fixes #201

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @devimc - looks fine, but please can you add some tests for `readAll()` (and ideally for `readCtl()` and `readTty()` too). It would be good if you could test a variety of length including `-1` (for `readAll()`), `0`, `1`, `4095`, `4096`, `4097`' and maybe a huge read of a few MB.

**Comment by devimc** [comment]
@jodh-intel I always forget to write unit tests 😄 , lemme write them 
thanks

**Comment by grahamwhaley** [comment]
heh heh - one way around that @devimc is to write the tests before the code ;-) [TDD](https://en.wikipedia.org/wiki/Test-driven_development)

**Comment by jodh-intel** [comment]
... which is odd because you are really good at writing them! :smile: Thanks.

**Comment by devimc** [comment]
@jodh-intel changes applied, thanks

File: agent_test.go [hunk 1]
```diff
@@ -153,3 +154,161 @@ func TestAgentConfigGetConfig(t *testing.T) {
...
+		// write command and length
+		buff = make([]byte, ctlHeaderSize)
+		binary.BigEndian.PutUint32(buff[:4], hCmd)
+		binary.BigEndian.PutUint32(buff[4:ctlHeaderSize], s+ctlHeaderSize)
+		n, err := f.Write(buff)
+		if err != nil {
+			t.Error(err)
+		}
+		if uint32(n) != uint32(ctlHeaderSize) {
+			t.Error("expected: %d, actual: %d", n, ctlHeaderSize)
```
Comment (jodh-intel) [inline agent_test.go]: Travis is failing as the linters have detected this and a few other calls should be `t.Errorf()`.

Comment (devimc) [reply agent_test.go]: oops!, fixed, thanks

**Review by jodh-intel — APPROVED** [review]

**Review by amshinde — APPROVED** [review]
lgtm

**Pull request merged**